lib/std/io/buffered_reader.zig

```diff
@@ -39,6 +39,25 @@ pub fn BufferedReader(comptime buffer_size: usize, comptime ReaderType: type) ty
             return dest.len;
         }
 
+        pub fn peek(self: *Self, dest: []u8) !usize {
```

Comment:
* what does the return value mean?
* is there a way to ensure that some minimum amount will always be available for peeking?
* what happens if the buffer size is too small for the desired peek amount?

(answer with documentation comments please)